src/sentry/api/serializers/models/activity.py

```diff
@@ -87,6 +87,12 @@ def serialize(self, obj, attrs, user):
             data = {"fingerprints": obj.data["fingerprints"], "destination": attrs["destination"]}
         else:
             data = obj.data
+            # XXX: We had a problem where Users were embedded into the mentions
+            # attribute of group notes which needs to be removed
+            # While group_note update has been fixed there are still many skunky comments
+            # in the database.
+            if "mentions" in data:
```

Comment:
I'd just do the same `data.pop('mentions', None)` here too. Not really a reason to guard it against an `if`.